File: testing/PIMS.Tests.Automation/PageObjects/HelpDesk.cs
using OpenQA.Selenium;

namespace PIMS.Tests.Automation.PageObjects
{
    public class HelpDesk : PageObjectBase
    {
        //Help Desk Main Button
        private readonly By mainMenuHelpDeskBttn = By.CssSelector("header div[class='nav-item'] div");

        //Modal Header
        private readonly By mainMenuHeader = By.CssSelector("div[class='modal-title h4']");

        //Modal Content
        private readonly By helpDeskSubtitle = By.CssSelector("div[class='modal-body'] h3:first-child");
        private readonly By helpDeskDescription = By.XPath("//p[contains(text(),'This overview has useful tools that will support you to start using the application. You can also watch the video demos.')]");
        private readonly By helpDeskPIMSResourcesLink = By.XPath("//a[contains(text(),'PIMS Training Materials')]");

        private readonly By helpDeskContactUsSubtitle = By.XPath("//h3[contains(text(),'Contact us')]");
        private readonly By helpDeskUserLabel = By.XPath("//label[contains(text(),'Name')]");
        private readonly By helpDeskUserInput = By.Id("input-user");
        private readonly By helpDeskEmailLabel = By.XPath("//label[contains(text(),'Email')]");
        private readonly By helpDeskEmailInput = By.Id("input-email");
        private readonly By helpDeskDescriptionLabel = By.XPath("//div[@class='modal-body']/div/form/div/div/label[contains(text(),'Description')]");
        private readonly By helpDeskDescriptionInput = By.Id("input-description");
        private readonly By helpDeskFooterQuestion = By.XPath("//p[contains(text(),'Do you want to proceed and send the email?')]");

        //Help Desk Buttons Elements
        private readonly By noButton = By.CssSelector("button[data-testid='cancel-modal-button']");
        private readonly By yesButton = By.CssSelector("div[class='modal-footer'] a[data-testid='ok-modal-button']");

        public HelpDesk(IWebDriver webDriver) : base(webDriver)
        {}

        public void NavigateToHelpDesk()
        {
            Wait();
            FocusAndClick(mainMenuHelpDeskBttn);
        }

        public void VerifyHelpDeskModal()
        {
            Wait();

            AssertTrueContentEquals(mainMenuHeader, "Help Desk");

            AssertTrueContentEquals(helpDeskSubtitle, "Get started with PIMS");
            AssertTrueIsDisplayed(helpDeskDescription);
            AssertTrueIsDisplayed(helpDeskPIMSResourcesLink);

            AssertTrueIsDisplayed(helpDeskContactUsSubtitle);
            AssertTrueIsDisplayed(helpDeskUserLabel);
            AssertTrueIsDisplayed(helpDeskUserInput);
            AssertTrueIsDisplayed(helpDeskEmailLabel);
            AssertTrueIsDisplayed(helpDeskEmailInput);
            AssertTrueIsDisplayed(helpDeskDescriptionLabel);                           
            AssertTrueIsDisplayed(helpDeskDescriptionInput);
            AssertTrueIsDisplayed(helpDeskFooterQuestion);

            AssertTrueIsDisplayed(noButton);
            AssertTrueIsDisplayed(yesButton);
        }

        public void CancelHelpDeskModal()
        {
            WaitUntilClickable(noButton);
            webDriver.FindElement(noButton).Click();
        }
    }
}

File: testing/playwright-testing/support/pages/HelpDesk.js
const { expect } = require("@playwright/test");
const { clickAndWaitFor } = require("../../support/common.js");

class HelpDesk {
  constructor(page) {
    this.page = page;
  }

  async navigateHelpDeskForm() {
    clickAndWaitFor(
      this.page,
      "header div[class='nav-item'] div",
      "div[class='modal-content']"
    );
  }

  async verifyHelpDeskModal() {
    const helpDeskTitle =
      (
        await this.page.locator("div[class='modal-title h4']").textContent()
      )?.trim() || "";
    expect(helpDeskTitle).toEqual("Help Desk");

    const helpDeskSubtitle =
      (
        await this.page
          .locator("div[class='modal-body'] h3:first-child")
          .textContent()
      )?.trim() || "";
    expect(helpDeskSubtitle).toEqual("Get started with PIMS");

    const helpDeskInstructions = await this.page
      .locator("div[class='modal-body'] p")
      .first()
      .textContent();
    expect(helpDeskInstructions).toEqual(
      "This overview has useful tools that will support you to start using the application. You can also watch the video demos."
    );

    const helpDeskPIMSResourcesLink = await this.page.locator(
      "//a[contains(text(),'PIMS Training Materials')]"
    );
    expect(helpDeskPIMSResourcesLink).toBeVisible();

    const helpDeskContactUsSubtitle = await this.page.locator(
      "//h3[contains(text(),'Contact us')]"
    );
    expect(helpDeskContactUsSubtitle).toBeVisible();

    const helpDeskUserLabel = await this.page.locator(
      "//label[contains(text(),'Name')]"
    );
    expect(helpDeskUserLabel).toBeVisible();

    const helpDeskUserInput = await this.page.locator("#input-user");
    expect(helpDeskUserInput).toBeVisible();

    const helpDeskEmailLabel = await this.page.locator(
      "//label[contains(text(),'Email')]"
    );
    expect(helpDeskEmailLabel).toBeVisible();

    const helpDeskEmailInput = await this.page.locator("#input-email");
    expect(helpDeskEmailInput).toBeVisible();

    const helpDeskDescriptionLabel = await this.page.locator(
      "//div[@class='modal-body']/div/form/div/div/label[contains(text(),'Description')]"
    );
    expect(helpDeskDescriptionLabel).toBeVisible();

    const helpDeskDescriptionInput = await this.page.locator(
      "#input-description"
    );
    expect(helpDeskDescriptionInput).toBeVisible();

    const helpDeskFooterQuestion = await this.page
      .locator("div[class='modal-body'] p")
      .last()
      .textContent();
    expect(helpDeskFooterQuestion).toEqual(
      "Do you want to proceed and send the email?"
    );

    const noButton = await this.page.locator(
      "button[data-testid='cancel-modal-button']"
    );
    expect(noButton).toBeVisible();

    const yesButton = await this.page.locator(
      "div[class='modal-footer'] a[data-testid='ok-modal-button']"
    );
    expect(yesButton).toBeVisible();
  }

  async cancelHelpDeskModal() {
    await this.page
      .locator("button[data-testid='cancel-modal-button']")
      .click();
  }
}

module.exports = HelpDesk;
